helper.py

```diff
@@ -3,33 +3,39 @@ def getUser(conn, author):
     Query the User
     """
     c = conn.cursor()
-    c.execute("SELECT user_id, votes from votes WHERE user_id=?", (int(author), ))
+    c.execute("SELECT user_id, COUNT(*) as votes from votes_history WHERE user_id=?", (int(author), ))
     row = c.fetchone()
     return row
 
-def createUser(conn, author):
+def updateHistory(conn, author, message_id, backer):
     """
-    Creates the User
+    Updates the history
+    Returns success
     """
     c = conn.cursor()
-    c.execute("INSERT INTO votes (user_id, votes) VALUES (?,1)", (int(author), ))
+    c.execute("INSERT INTO votes_history (user_id, message_id, backer) VALUES (?,?,?)", (int(author), int(message_id), int(backer), ))
     conn.commit()
+    return c.rowcount > 0
 
-def updateVote(conn, author, vote):
+def removeHistory(conn, message_id, author = -1, backer = -1):
     """
-    Updates the Vote counter
+    Updates the history to remove vote
+    Returns success
     """
     c = conn.cursor()
-    c.execute("UPDATE votes SET votes=? WHERE user_id=?", (int(vote), int(author), ))
+    if backer != -1 and author != -1:
+        c.execute("DELETE FROM votes_history WHERE user_id=? AND message_id=? AND backer=?", (int(author), int(message_id), int(backer), ))
+    else:
+        c.execute("DELETE FROM votes_history WHERE message_id=?", (int(message_id), ))
     conn.commit()
-
-def updateHistory(conn, author, message_id, backer):
+    return c.rowcount > 0
+def fetchLeaderboard(conn):
```

Comment:
```suggestion

def fetchLeaderboard(conn):
```

Reply:
The return statement is for the possible later use case where there is a unique key to avoid duplicate entries.

At first I thought this could be used to avoid #1 but later found `message_id` to not be unique.

Would still leave it as confirmation.

Reply:
never mind misread the proposed change...